lib/vanagon/common/pathname.rb

```diff
@@ -28,7 +30,7 @@ class Pathname
       #   and exposed through the {#configfile?} method.
       # @return [Vanagon::Common::Pathname] Returns a new Pathname instance.
       def initialize(path, mode: nil, owner: nil, group: nil, config: false)
-        @path = File.expand_path(path)
+        @path = ::Pathname.new(path).cleanpath
```

Comment:
Did you determine if this needs to be coerced to a string or if the use of the bare `::Pathname` object was OK?